library/packages/src/lib/packages/repository.rb

```diff
@@ -0,0 +1,118 @@
+require "packages/repository_product"
+
+module Packages
+  class Repository
+    Yast.import "Pkg"
+    Yast.import "URL"
+    
+    # @return [Fixnum] Repository ID
+    attr_reader :repo_id
+    # @return [String] Repository name
+    attr_reader :name
+    # @return [String] Repository URL
+    attr_reader :url
+
+    # The repository scheme could not be determined
+    class UnknownScheme < StandardError; end
+
+    # Repository was not found
+    class NotFound < StandardError; end
+
+    class << self
+      # Return all registered repositories
+      #
+      # @see Yast::Pkg.SourceGetCurrent
+      # @see Packages::Repository.find
+      def all
+        Yast::Pkg.SourceGetCurrent(true).map do |repo_id|
+          find(repo_id)
+        end
+      end
+
+      # Return a repository with the given repo_id
+      #
+      # @param repo_id [Fixnum] Repository ID
+      # @return [Packages::Repository] Repository instance
+      #
+      # @raise NotFound
+      def find(repo_id)
+        repo_data = Yast::Pkg.SourceGeneralData(repo_id)
+        raise NotFound if repo_data.nil?
+        new(repo_id: repo_id, enabled: repo_data["enabled"],
+          name: repo_data["name"], autorefresh: repo_data["autorefresh"],
+          url: repo_data["url"])
+      end
+    end
+    
+    # Constructor
+    #
+    # @param repo_id     [Fixnum]  Repository ID
+    # @param name        [String]  Name
+    # @param enabled     [Boolean] Is the repository enabled?
+    # @param autorefresh [Boolean] Is auto-refresh enabled for this repository?
+    # @param url         [String]  Repository URL
+    def initialize(repo_id:, name:, enabled:, autorefresh:, url:)
+      @repo_id = repo_id
+      @name    = name
+      @enabled = enabled
+      @autorefresh = autorefresh
+      @url     = url
+    end
+
+    # Return repository scheme
+    #
+    # The scheme is determined using the URL
+    #
+    # @raise UnknownScheme
+    def scheme
+      return @scheme unless @scheme.nil?
+      data_from_url = Yast::URL.Parse(url)
+      raise UnknownScheme if data_from_url["scheme"].nil?
+      @scheme = data_from_url["scheme"].to_sym
+    end
+
+    # Return products contained in the repository
+    #
+    # @return [Array<Packages::RepositoryProduct>] Products in the repository
+    #
+    # @see Yast::Pkg.ResolvableProperties
+    # @see Packages::RepositoryProduct
+    def products
+      return @products unless @products.nil?
+
+      # Filter products from this repository
+      candidates = Yast::Pkg.ResolvableProperties("", :product, "").select do |pro|
+        pro["source"] == repo_id
+      end
+      
+      # Build an array of Packages::Product objects
+      @products = candidates.map do |data|
+        Packages::RepositoryProduct.new(name: data["name"], version: data["version"],
```

Comment:
Just a note: Some time ago I have added also the parts of the `version` attribute into the response, see https://github.com/yast/yast-pkg-bindings/pull/42.

But I think we can keep it simple for now, the version parts can be added later when needed.